### ISSUE TITLE
[Core] Fix named imports across the entire codebase

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket#884]
## Objective

Follow best Golang practices w.r.t named imports

## Origin Document

![Screenshot 2023-07-05 at 2 50 44 PM](https://github.com/pokt-network/pocket/assets/1892194/c63a023e-3662-477c-8cb9-a1345f035126)

## Goals

- Follow best practices in Go
- Stay consistent w.r.t imports throughout the codebase

## Deliverable

- [ ] Identify and update all named imports in Go that are `CamelCase` and change them to `snake_case`

## Non-goals / Non-deliverables

- Any business logic changes

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
- [ ] Update any relevant local/global README(s)
- [ ] Update relevant source code tree explanations
- [ ] Add or update any relevant or supporting [mermaid](https://mermaid-js.github.io/mermaid/) diagrams

## Testing Methodology

- [ ] **Task specific tests or benchmarks**: `make ...`
- [ ] **New tests or benchmarks**: `make ...`
- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)
- [ ] **k8s LocalNet**: verify a `k8s LocalNet` is still functioning correctly by following the instructions [here](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md)

---

**Creator**: @Olshansk

**Comment by Olshansk** [comment]
@kutoft We discussed you trying out a `core starter task` issue to get "hands on" with the codebase. Wdyt of this one?

**Comment by Olshansk** [comment]
@kutoft Do you want to try and tackle this in the upcoming iteration?

**Comment by Olshansk** [comment]
_tl;dr Goal is to make all named imports `snake_case`_

(2) is what we **DON'T WANT** and (1) is what we **DO WANT**
---

1. If we do a regex search all `camelCase` imports (e.g. `^\s*([a-z]+(?:[A-Z][a-z]*)*)\s`), we find the following:

![Screenshot 2023-07-31 at 2 19 22 PM](https://github.com/pokt-network/pocket/assets/1892194/1a12ee23-0c4e-45fb-884f-2662ccce9364)

2. If we do a regex search all `snake_case` imports (e.g. `^\s*([a-z]+_[a-z_]+)\s+"github\.com(.*?)"`), we find the following:

![Screenshot 2023-07-31 at 2 20 17 PM](https://github.com/pokt-network/pocket/assets/1892194/139011ee-aa83-4a3c-9086-834f2117a308)

**Comment by kutoft** [comment]
@Olshansk Your message seems a little contradictory. So I just want to confirm which casing structure you want here, camelCase or snake_case.

Additionally, I was reading up on GO language patterns, and it appears the common practice is to name packages as all lowercase. Would you rather stick with this convention?

https://go.dev/blog/package-names

**Comment by Olshansk** [comment]
Response to the question is below but `please hold off on implementation until further notice` while our team reevaluated priorities.

---

> Additionally, I was reading up on GO language patterns, and it appears the common practice is to name packages as all lowercase. Would you rather stick with this convention?

Yes. Attaching the GPT screenshot from our offline conversation for reference.

![ChatGPT-Go_Naming_Conventions](https://github.com/pokt-network/pocket/assets/1892194/2bbe798d-06fa-436c-9fe8-27664472f1a9)

Tl;dr

1. Replace `snake_case` with `lowercase`
2. Replace `PascalCase` with `lowercase`
3. Replace `camelCase` with `lowercast`
